src/aws/auth.rs

```diff
@@ -133,13 +150,37 @@ impl AwsAuthentication {
             Self::AccessKey {
                 access_key_id,
                 secret_access_key,
-            } => Ok(SharedCredentialsProvider::new(Credentials::from_keys(
-                access_key_id.inner(),
-                secret_access_key.inner(),
-                None,
-            ))),
-            AwsAuthentication::File { .. } => {
-                Err("Overriding the credentials file is not supported.".into())
+                assume_role,
+                region,
+            } => {
+                let provider = SharedCredentialsProvider::new(Credentials::from_keys(
+                    access_key_id.inner(),
+                    secret_access_key.inner(),
+                    None,
+                ));
+                if let Some(assume_role) = assume_role {
+                    let auth_region = region.clone().map(Region::new).unwrap_or(service_region);
+                    let provider = AssumeRoleProviderBuilder::new(assume_role)
+                        .region(auth_region)
+                        .build(provider);
+                    return Ok(SharedCredentialsProvider::new(provider))
```

Comment:
```suggestion
                    Ok(SharedCredentialsProvider::new(provider))
```

I think we can drop the `return`?

Reply:
The return's still needed, otherwise the `Ok()` inside the if-let block would be unused. Worse, it wouldn't compile overall because the outer `Ok(provider)` is trying to return a `provider` that had been moved inside the if-let block.

Reply:
Ah, 👍 gooood point.